module/VuFind/tests/unit-tests/src/VuFindTest/RecordDriver/EDSTest.php

```diff
@@ -376,21 +536,32 @@ protected function getDriverWithIdentifierData(): EDS
                         ],
                     ],
                 ],
-            ]
+            ],
+            $config
         );
     }
 
     /**
      * Get a record driver with fake data.
      *
      * @param array $overrides Raw data for testing
+     * @param array $config    overrides the EDS config for testing
      *
      * @return EDS
      */
-    protected function getDriver($overrides = []): EDS
+    protected function getDriver($overrides = [], array $config = null): EDS
     {
         $record = new EDS();
         $record->setRawData($overrides);
+
+        // Set the private recordConfig property
+        // TODO -- is there a better way to set this config?
+        $this->setProperty(
```

Comment:
Good news! There's a very easy way to do this more cleanly -- you can pass the configuration in through the record driver constructor, so up above, you can just change the first line of this method to:

`$record = new EDS(null, new \Laminas\Config\Config($config ?? $this->defaultDriverConfig))`

Also note that by building a valid Laminas Config object, I think you can keep the existing EDS code as-is instead of having to do the `(array)` typecasting introduced above, which might behave less predictably.

Reply:
Thanks, that did the trick. I knew there had to be a cleaner way of doing it that I wasn't looking deep enough for! And that did indeed allow me to remove the changes I had made in the EDS driver.

This should be ready for review now. Let me know if you have any other suggestions to further improve the tests!